### PR TITLE
Enable eager mode tests, fix mixed ops and improve coverage

### PR DESCRIPTION
**Pull request by klecki**
## Description:
Eager mode test coverage was not run - add it to the test script. Add missing test cases to cover the new operators, disable some operators according to the TODO: only CPU tests are now supported.

Fix few problematic test-cases.

Fix mixed ops in the eager mode that were not working due to refactoring.

Expose new stateful ops in the eager API.

As a followup, deduplication in test_eager_coverage and test_dali_cpu_only should be considered.

### Affected modules and functionalities:
Eager mode & tests

### Tests:

- [x] Existing tests apply 
(now they do)
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/python/nvidia/dali/_utils/eager_utils.py [hunk 1]
```diff
@@ -209,6 +213,9 @@ def _slice_tensorlist(data, size):
     return type(data)(list(data)[:size], layout=data.layout())
 
 
+# TODO(klecki): Expose `nvidia.dali.math` in eager mode
+
+
```
Comment (szalpal) [inline dali/python/nvidia/dali/_utils/eager_utils.py:218]: I have reservations with leaving a TODO in code. Does that mean anything? ;)

**Review by szalpal — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Comment by klecki** [comment]
!build

File: dali/python/nvidia/dali/_utils/eager_utils.py (hunk 1 above)
Comment (klecki) [reply dali/python/nvidia/dali/_utils/eager_utils.py:218]: I hope. This is experimental module which we currently deprioritized, so maybe if we pick it up again we will clear this one.

**Pull request merged**